File: C++ codes/Wall.cpp
#include<ctime>
#include"Wall.h"

Wall::Wall(User *owner){
	this->owner = owner;
}

void Wall::post(User *from, string message){
    Message *post = new Message(from, message);
    messages.push_back(post);
}

vector<Message*> Wall::getPosts(){
	return messages;
}

User* Wall::getUserName(){
	return owner;
}

File: JAVA codes/Wall.java
import java.util.ArrayList;

public class Wall 
{
    private User owner;//o xrhsths pou exei to wall auto 
    private ArrayList<Message> messages;//to melos messages einai typou arraylist<message>kai periexei ola ta mynhmata pou einai dhmosieumena sto wall tou
    
    public Wall(User owner)
    {
        this.owner = owner;
        messages = new ArrayList<>();
    }
    
    public void post(User from, String message)
    {
        Message post = new Message(from, message);
        messages.add(post);
    }
   
    public ArrayList<Message> getPosts()//epistrefei to arraylist me ola ta messages pou vriskontai sto wall autou tou xrhsth
    {
        return messages;
    }
    
    public User getUserName()
    {
        return owner;
    }
}